Packs/DuoAdminApi/Integrations/DuoEventCollector/DuoEventCollector.py

```diff
@@ -38,53 +38,117 @@ class Client:
-        self.params = params.get('params')  # type: ignore[attr-defined]
-        self.admin_api = create_api_call(params.get('host'),  # type: ignore[attr-defined]
-                                         params.get('integration_key'),  # type: ignore[attr-defined]
-                                         (params.get('secret_key')).get('password'))  # type: ignore[attr-defined]
+    def __init__(self, params: Params):
+        self.params = params
+        self.admin_api = create_api_call(self.params.host,
+                                         self.params.integration_key,
+                                         str(self.params.secret_key.get('password')))
 
-    def call(self, request_order: list) -> dict:  # pragma: no cover
+    def call(self, request_order: list) -> tuple:
         retries = int(self.params.retries)
-        auth_api_version = int(self.params.auth_api_version)
+        response_metadata = None
         while retries != 0:
             try:
                 if request_order[0] == LogType.AUTHENTICATION:
-                    if auth_api_version == 2:
-                        response = self.admin_api.get_authentication_log(
-                            mintime=self.params.mintime[LogType.AUTHENTICATION] * AUTH_API_V2_TIME_MODIFIER,
-                            api_version=2, limit='1000')
-                        # The v2 API returns a different object, so we do this to normalize it
-                        response = response.get('authlogs', [])
-                    elif auth_api_version == 1:
-                        response = self.admin_api.get_authentication_log(
-                            mintime=self.params.mintime[LogType.AUTHENTICATION])
-                elif request_order[0] == LogType.ADMINISTRATION:
-                    response = self.admin_api.get_administrator_log(
-                        mintime=self.params.mintime[LogType.ADMINISTRATION])
+                    events, response_metadata = self.handle_authentication_logs()
+
                 elif request_order[0] == LogType.TELEPHONY:
-                    response = self.admin_api.get_telephony_log(
-                        mintime=self.params.mintime[LogType.TELEPHONY])
-                return response
+                    events = self.handle_telephony_logs_v1()
+
+                elif request_order[0] == LogType.ADMINISTRATION:
+                    events = self.handle_administration_logs()
+
+                events = parse_events(events)
+                return events, response_metadata
+
             except Exception as exc:
                 msg = f'something went wrong with the sdk call {exc}'
                 demisto.debug(msg)
                 if str(exc) == 'Received 429 Too Many Requests':
                     retries -= 1
                 else:
-                    retries = 0
-        return {}
+                    raise exc
+
+        return ([], [])
+
+    def handle_authentication_logs(self) -> tuple:
```

Comment:
The term "handle" is very ambiguous. You're getting the auth logs, so `get_authentication_logs`.